pkg/instance/instance.go

```diff
@@ -154,66 +157,44 @@ func (i *Instance) makeVFS() error {
 	return err
 }
 
-// AppsFS returns the hidden filesystem associated with the specified
+// AppsCopier returns the application copier associated with the specified
 // application type
-func (i *Instance) AppsFS(appsType apps.AppType) afero.Fs {
+func (i *Instance) AppsCopier(appsType apps.AppType) apps.Copier {
+	fsURL := config.FsURL()
+	var baseDirName string
 	switch appsType {
 	case apps.Webapp:
-		return i.hiddenFS(vfs.WebappsDirName)
+		baseDirName = vfs.WebappsDirName
 	case apps.Konnector:
-		return i.hiddenFS(vfs.KonnectorsDirName)
+		baseDirName = vfs.KonnectorsDirName
 	}
-	panic(fmt.Errorf("Unknown application type %s", string(appsType)))
+	baseFS := afero.NewBasePathFs(afero.NewOsFs(),
+		path.Join(fsURL.Path, i.Domain, baseDirName))
+	return apps.NewAferoCopier(baseFS)
+}
+
+// AppsFileServer returns the web-application file server associated to this
+// instance.
+func (i *Instance) AppsFileServer() apps.FileServer {
+	fsURL := config.FsURL()
+	baseFS := afero.NewBasePathFs(afero.NewOsFs(),
+		path.Join(fsURL.Path, i.Domain, vfs.WebappsDirName))
+	return apps.NewAferoFileServer(baseFS, nil)
 }
 
 // ThumbsFS returns the hidden filesystem for storing the thumbnails of the
 // photos/image
 func (i *Instance) ThumbsFS() afero.Fs {
-	return i.hiddenFS(vfs.ThumbsDirName)
-}
-
-func (i *Instance) hiddenFS(dirname string) afero.Fs {
 	fsURL := config.FsURL()
-	switch fsURL.Scheme {
-	case "file", "mem":
-		return afero.NewBasePathFs(afero.NewOsFs(),
-			path.Join(fsURL.Path, i.Domain, dirname))
-	case "swift":
-		panic("Not implemented")
-	}
-	return nil
+	return afero.NewBasePathFs(afero.NewOsFs(),
+		path.Join(fsURL.Path, i.Domain, vfs.ThumbsDirName))
```

Comment:
Currently, we still support only local file system for apps, konnectors, and thumbs. Right?